### PR TITLE
ML extension - Improving predict parameter mapping experience 

### PR DESCRIPTION
**Pull request by llali**
Predict input/output page
 - Added validation to predict parameter mapping page 
 - Not preselecting table and database to make sure user select the right one 
 - Showing a warning if data types don't match

Bug fixes:
- checking for prediction dependency before opening the Wizard
- checking the version ot server to make sure onnx is supported
- added tooltip
- fixed an issue with loading models from database 
- importing model description and framework from Azure

### CODE REVIEW

File: extensions/machine-learning/src/views/widgets/dashboardWidget.ts [hunk 1]
```diff
@@ -506,7 +506,7 @@ export class DashboardWidget {
 		const iconContainer = view.modelBuilder.flexContainer().withLayout({
 			flexFlow: 'row',
 			width: maxWidth,
-			height: maxHeight - 20,
+			height: maxHeight - 23,
```
Comment (Benjin) [inline extensions/machine-learning/src/views/widgets/dashboardWidget.ts:509]: magic constant?  maybe define so there's a named variable, or add a comment

File: extensions/machine-learning/src/test/views/models/ModelManagementController.test.ts [hunk 2]
```diff
@@ -125,16 +126,24 @@ describe('Model Controller', () => {
...
 		testContext.predictService.setup(x => x.getTableList(TypeMoq.It.isAny())).returns(() => Promise.resolve(tableNames));
 		testContext.azureModelService.setup(x => x.getAccounts()).returns(() => Promise.resolve(accounts));
+		testContext.azureModelService.setup(x => x.signInToAzure()).returns(() => Promise.resolve());
 		testContext.azureModelService.setup(x => x.getSubscriptions(TypeMoq.It.isAny())).returns(() => Promise.resolve(subscriptions));
 		testContext.azureModelService.setup(x => x.getGroups(TypeMoq.It.isAny(), TypeMoq.It.isAny())).returns(() => Promise.resolve(groups));
 		testContext.azureModelService.setup(x => x.getWorkspaces(TypeMoq.It.isAny(), TypeMoq.It.isAny(), TypeMoq.It.isAny())).returns(() => Promise.resolve(workspaces));
 		testContext.azureModelService.setup(x => x.getModels(TypeMoq.It.isAny(), TypeMoq.It.isAny(), TypeMoq.It.isAny(), TypeMoq.It.isAny())).returns(() => Promise.resolve(models));
+		testContext.azureModelService.setup(x => x.downloadModel(TypeMoq.It.isAny(), TypeMoq.It.isAny(), TypeMoq.It.isAny(), TypeMoq.It.isAny(), TypeMoq.It.isAny())).returns(() => Promise.resolve('path'));
```
[the remark this answers is not in the document]
Comment (llali) [reply extensions/machine-learning/src/test/views/models/ModelManagementController.test.ts:144]: I tried to do that as much as I could but some tests have to setup to specific data

File: extensions/machine-learning/src/test/views/models/predictWizard.test.ts [hunk 3]
```diff
@@ -9,7 +9,7 @@ import 'mocha';
 import { createContext } from './utils';
 import {
 	ListModelsEventName, ListAccountsEventName, ListSubscriptionsEventName, ListGroupsEventName, ListWorkspacesEventName,
```
[the remark this answers is not in the document]
Comment (llali) [reply extensions/machine-learning/src/test/views/models/predictWizard.test.ts:11]: the best practice is to import specific one if don't need all modules

File: extensions/machine-learning/src/modelManagement/interfaces.ts [hunk 4]
```diff
@@ -18,6 +18,7 @@ export interface ListWorkspaceModelsResult extends Array<WorkspaceModel> {
  */
 export interface WorkspaceModel extends Resource {
 	framework?: string;
```
Comment (udeeshagautam) [inline extensions/machine-learning/src/modelManagement/interfaces.ts:20]: nothing for this PR per se but it seems bit odd that this interface has no required types... is that intensional?

File: extensions/machine-learning/src/modelManagement/modelConfigRecent.ts [hunk 5]
```diff
@@ -21,7 +21,12 @@ export class ModelConfigRecent {
 	}
 
 	public storeModelTable(connection: azdata.connection.ConnectionProfile, table: DatabaseTable): void {
-		this._memento.update(this.getKey(connection), table);
+		if (connection && table?.databaseName && table?.tableName && table?.schema) {
```
Comment (udeeshagautam) [inline extensions/machine-learning/src/modelManagement/modelConfigRecent.ts:24]: may be && table &&... rest of the checks?

File: extensions/machine-learning/src/test/views/models/registerModelWizard.test.ts [hunk 6]
```diff
@@ -7,24 +7,78 @@ import * as azdata from 'azdata';
...
+import { ListModelsEventName, ListAccountsEventName, ListSubscriptionsEventName, ListGroupsEventName, ListWorkspacesEventName, ListAzureModelsEventName, ModelSourceType, ListDatabaseNamesEventName, ListTableNamesEventName, VerifyImportTableEventName } from '../../../views/models/modelViewBase';
 import { ImportedModel } from '../../../modelManagement/interfaces';
 import { azureResource } from '../../../typings/azure-resource';
 import { Workspace } from '@azure/arm-machinelearningservices/esm/models';
 import { ViewBase } from '../../../views/viewBase';
 import { WorkspaceModel } from '../../../modelManagement/interfaces';
 import { ImportModelWizard } from '../../../views/models/manageModels/importModelWizard';
+import { DatabaseTable } from '../../../prediction/interfaces';
 
+let accounts: azdata.Account[] = [
```
Comment (udeeshagautam) [inline extensions/machine-learning/src/test/views/models/registerModelWizard.test.ts:19]: nothing to do with this R but this reminds me that when I tried the AML flow it was not able to remember anything about azure account - can that be made better? let me know if you have something on that or should i open a followup task

File: extensions/machine-learning/src/test/views/utils.ts [hunk 7]
```diff
@@ -240,7 +240,7 @@ export function createViewContext(): ViewTestContext {
 			try {
 				await handler(view);
 			} catch (err) {
-				throw err;
+				console.log(err);
```
Comment (udeeshagautam) [inline extensions/machine-learning/src/test/views/utils.ts:243]: what will change for user experience?

File: extensions/machine-learning/src/views/models/localModelsComponent.ts [hunk 8]
```diff
@@ -65,7 +62,7 @@ export class LocalModelsComponent extends ModelViewBase implements IDataComponen
 			.withLayout({
 				flexFlow: 'row',
 				justifyContent: 'space-between',
-				width: this.componentMaxLength
+				width: this.componentMaxLength + 200
```
Comment (udeeshagautam) [inline extensions/machine-learning/src/views/models/localModelsComponent.ts:65]: if you could add a comment as to why

**Pull request merged**